### PR TITLE
Bluetooth: Mesh: Trim _USES_MBEDTLS_PSA selects

### PR DESCRIPTION
**Pull request by HaavardRei**
Removes legacy configuration options from BT_MESH_USES_MBEDTLS_PSA. The same coverage is provided by existing PSA_WANT_... selects.

### CODE REVIEW

**Review by PavelVPV — DISMISSED** [review]

**Review by valeriosetti — DISMISSED** [review]

**Review by alxelax — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

File: subsys/bluetooth/mesh/Kconfig [hunk 1]
```diff
@@ -1477,25 +1477,21 @@ config BT_MESH_USES_MBEDTLS_PSA
...
 	select PSA_WANT_ALG_ECB_NO_PADDING
 	select PSA_WANT_KEY_TYPE_AES
-	select MBEDTLS_CIPHER_AES_ENABLED
 	select MBEDTLS_AES_ROM_TABLES
 	select PSA_WANT_ALG_CCM
 	select PSA_WANT_KEY_TYPE_HMAC
 	select PSA_WANT_ALG_HMAC
 	select PSA_WANT_ALG_SHA_256
 	select PSA_WANT_ALG_ECDH
 	select PSA_WANT_ALG_ECDSA
```
Comment (alxelax) [inline subsys/bluetooth/mesh/Kconfig:1493]: I guess `PSA_WANT_ALG_ECDSA` is not required either. This is for signatures those Mesh doesn't use. Previously, mbedtls united it with ECDH.

Comment (HaavardRei) [reply subsys/bluetooth/mesh/Kconfig]: Removed it now

**Review by valeriosetti — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Comment by jhedberg** [comment]
Looks like there's some unrelated CI breakage: 
```
INFO: Error found: storage.stream_flash on nrf52840dk/nrf52840 (Not in testsuite platform allow list but is one of the integration platforms)
```

I've seen the same failure in other PRs too.

**Comment by jhedberg** [comment]
> Looks like there's some unrelated CI breakage:
> 
> ```
> INFO: Error found: storage.stream_flash on nrf52840dk/nrf52840 (Not in testsuite platform allow list but is one of the integration platforms)
> ```
> 
> I've seen the same failure in other PRs too.

Hotfix available here: #82222

**Review by alxelax — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**